file_transfer_agent_test.go

```diff
@@ -846,21 +853,26 @@ func TestReadonlyTmpDirPathShouldFail(t *testing.T) {
 	if err != nil {
 		t.Fatalf("cannot create temp directory: %v", err)
 	}
-	defer os.RemoveAll(tmpDir)
+	defer func() {
+		assertNilF(t, os.RemoveAll(tmpDir))
+	}()
 
 	uploadFile := filepath.Join(tmpDir, "data.txt")
 	f, err := os.Create(uploadFile)
 	if err != nil {
 		t.Error(err)
 	}
-	f.WriteString("test1,test2\ntest3,test4\n")
-	f.Close()
+	_, err = f.WriteString("test1,test2\ntest3,test4\n")
+	assertNilF(t, err)
+	assertNilF(t, f.Close())
 
-	err = os.Chmod(tmpDir, 0400)
+	err = os.Chmod(tmpDir, 0500)
 	if err != nil {
 		t.Fatalf("cannot mark directory as readonly: %v", err)
```

Comment:
This message should be updated

Reply:
Actually this is still valid. The only case is that we need `-x` to not only read files in a dir, but also to access files inside it. But the main case is still - this dir is readonly, not writable.